lib/vsc/mympirun/factory.py

```diff
@@ -51,16 +57,19 @@ def getinstance(mpi, sched, options):
     cache_key = (mpi, sched)
 
     if cache_key not in _coupler_class_cache:
-        class Coupler(mpi, sched):
-            """Temporary class to couple MPI and local sched"""
-            def __init__(self, **kwargs):
-                self.log = fancylogger.getLogger("%s_%s" % (mpi.__name__, sched.__name__))
-                super(Coupler, self).__init__(**kwargs)
-
-        _coupler_class_cache[cache_key] = Coupler
-        _log.debug("Created new 'Coupler' class for %s: %s", cache_key, id(Coupler))
-
-    coupler_class = _coupler_class_cache[cache_key]
-    _log.debug("Fetched Coupler class from cache %s: %s", cache_key, id(coupler_class))
+        coupler_class = type(
+            "Coupler_%s_%s" % (mpi.__name__, sched.__name__),
+            cache_key,
+            {
+                '__metaclass__': CouplerMetaClass,
```

Comment:
This is going to be fun to port to Py2 + Py3...

Reply:
i'll get rid of the metaclass, it's a very complicated way to init a log instance